### PR TITLE
IBX-5309 - Payment search criteria and sort clauses doc

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | [IBX-5309](https://issues.ibexa.co/browse/IBX-5309) |
| Versions      | 4.5 |
| Edition       | Commerce |

Document search criteria and sort clauses used in Payment management

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date

### CODE REVIEW

**Review by juskora — DISMISSED** [review]

File: docs/search/criteria_reference/payment_createdat_criterion.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+description: Payment CreatedAt Criterion
+edition: commerce
+---
+
+# Payment CreatedAt Criterion
+
+The `CreatedAtCriterion` Search Criterion searches for payments based on the date when they were initiated.
```
[the remark this answers is not in the document]
Comment (dabrt) [reply docs/search/criteria_reference/payment_createdat_criterion.md:8]: Let's see what Adam says, ok?

File: docs/search/criteria_reference/payment_createdat_criterion.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- `createdAt` - date to be matched, provided as a `DateTimeInterface` object
+- `operator` - optional operator constant (EQ, GT, GTE, LT, LTE)
+
+## Example
+
+``` php
+$criteria = new Ibexa\Contracts\Payment\Value\Payment\Query\Criterion\CreatedAt(
+    '2022-07-11T00:00:00+02:00',
+    Ibexa\Contracts\Payment\Value\Payment\Query\Criterion\Operator::GTE
```
Comment (adamwojs) [inline docs/search/criteria_reference/payment_createdat_criterion.md:20]: There is no such class as `Ibexa\Contracts\Payment\Value\Payment\Query\Criterion\Operator`

File: docs/search/criteria_reference/payment_currency_criterion.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+The `Currency` Search Criterion searches for payments based on the currency code.
+
+## Arguments
+
+- `currency` - string that represents a currency code
+
+## Example
+
+``` php
+$query->query = new Ibexa\Contracts\Payment\Value\Payment\Query\Criterion\Currency('EUR');
```
Comment (adamwojs) [inline docs/search/criteria_reference/payment_currency_criterion.md:17]:
```suggestion
$query->query = new \Ibexa\Contracts\Payment\Payment\Query\Criterion\Currency('EUR');
```

**Review by adamwojs — CHANGES_REQUESTED** [review]
I'm sorry but current state of this PR is unacceptable: 

- Code snippets contains references to non existing classes or other symbols
- Code snippets are inconsistent. Sometimes we refer to criteria using  FQCN sometime we do not. Formatting is no applied  
- Arguments passed to criteria have invalid type e.g. CreatedAt criterion expects DateTimeInterface but we are passing string

This needs to be fixed before we proceed to next step of code review. 

You find correct usage examples in integration tests
* https://github.com/ibexa/payment/blob/65cfb7a8caf20fea1898300a1fe6dcec7b4ab06b/tests/integration/PaymentServiceTest.php#L300-L495
* https://github.com/ibexa/payment/blob/c51cfdf22311af8b593e8a76d12d8d07fb7fe71d/tests/integration/PaymentMethodServiceTest.php#L251-L455

**Comment by DominikaK** [comment]
It seems the incorrect datetime examples were based on https://doc.ibexa.co/en/master/search/criteria_reference/createdat_criterion/#example, which contains the same error, will be fixed, but outside the scope of this PR.

File: docs/search/criteria_reference/payment_createdat_criterion.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dabrt) [reply docs/search/criteria_reference/payment_createdat_criterion.md]: @adamwojs Would you say initiated or created?

**Comment by dabrt** [comment]
@adamwojs please re-review the PR

Another question, out of the blue - don't we want to sort payments by their value? Being a financial manager I might find it useful to be able to filter out payments for a certain amount.

File: docs/search/criteria_reference/payment_createdat_criterion.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+# Payment CreatedAt Criterion
+
+The `CreatedAt` Search Criterion searches for payments based on the date when they were initiated.
+
+## Arguments
+
+- `createdAt` - date to be matched, provided as a `DateTimeInterface` object
+- `operator` - optional operator constant (EQ, GT, GTE, LT, LTE)
```
Comment (dabrt) [inline docs/search/criteria_reference/payment_createdat_criterion.md:13]:
```suggestion
- `operator` - optional operator string (EQ, GT, GTE, LT, LTE)
```

File: docs/search/criteria_reference/payment_method_createdat_criterion.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+# Payment Method CreatedAt Criterion
+
+The `CreatedAt` Search Criterion searches for payment methods based on the date when they were created.
+
+## Arguments
+
+- `createdAt` - date to be matched, provided as a `DateTimeInterface` object
+- `operator` - optional operator constant (EQ, GT, GTE, LT, LTE)
```
Comment (dabrt) [inline docs/search/criteria_reference/payment_method_createdat_criterion.md:13]:
```suggestion
- `operator` - optional operator string (EQ, GT, GTE, LT, LTE)
```

File: docs/search/criteria_reference/payment_method_updatedat_criterion.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+# Payment Method UpdatedAt Criterion
+
+The `UpdatedAt` Search Criterion searches for payment methods based on the date when their status was updated.
+
+## Arguments
+
+- `updatedAt` - date to be matched, provided as a `DateTimeInterface` object
+- `operator` - optional operator constant (EQ, GT, GTE, LT, LTE)
```
Comment (dabrt) [inline docs/search/criteria_reference/payment_method_updatedat_criterion.md:13]:
```suggestion
- `operator` - optional operator string (EQ, GT, GTE, LT, LTE)
```

File: docs/search/criteria_reference/payment_method_search_criteria.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+---
+description: Payment Method Search Criteria
+edition: commerce
+---
+
+# Payment Method Search Criteria reference
+
+Payment Method Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Payment Method Search.
+
+With these criteria you can filter payment methods by their payment method identifier, payment method creation date, payment method type, status, and so on.
```
Comment (DominikaK) [inline docs/search/criteria_reference/payment_method_search_criteria.md:10]:
```suggestion
With these Criteria you can filter payment methods by their payment method identifier, payment method creation date, payment method type, status, and so on.
```

File: docs/search/criteria_reference/payment_method_search_criteria.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+## Payment method search criteria
```
Comment (DominikaK) [inline docs/search/criteria_reference/payment_method_search_criteria.md:12]:
```suggestion
## Payment method Search Criteria
```

File: docs/search/criteria_reference/payment_search_criteria.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+---
+description: Payment Search Criteria
+edition: commerce
+---
+
+# Payment Search Criteria reference
+
+Payment Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Payment Search.
+
+With these criteria you can filter payments by their payment identifier, payment creation date, payment status, payment method, order, and so on.
```
Comment (DominikaK) [inline docs/search/criteria_reference/payment_search_criteria.md:10]:
```suggestion
With these Criteria you can filter payments by their payment identifier, payment creation date, payment status, payment method, order, and so on.
```

File: docs/search/criteria_reference/payment_search_criteria.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+## Payment search criteria
```
Comment (DominikaK) [inline docs/search/criteria_reference/payment_search_criteria.md:12]:
```suggestion
## Payment Search Criteria
```

File: docs/search/sort_clause_reference/payment_method_sort_clauses.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+---
+description: Payment Method Sort Clauses
+edition: commerce
+---
+
+---
```
Comment (DominikaK) [inline docs/search/sort_clause_reference/payment_method_sort_clauses.md:6]:
```suggestion
```

File: docs/search/sort_clause_reference/payment_sort_clauses.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
+---
+description: Payment Sort Clauses
+edition: commerce
+---
+
+---
```
Comment (DominikaK) [inline docs/search/sort_clause_reference/payment_sort_clauses.md:6]:
```suggestion
```

**Review by DominikaK — APPROVED** [review]

File: docs/search/criteria_reference/payment_createdat_criterion.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (DominikaK) [reply docs/search/criteria_reference/payment_createdat_criterion.md]: I'd opt for "created" as well. To me "initiated" refers more to a payment in a banking system, but here we're talking about creating a payment (object) inside our system

File: docs/search/criteria_reference/payment_createdat_criterion.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
+---
+description: Payment CreatedAt Criterion
+edition: commerce
+---
+
+# Payment CreatedAt Criterion
+
+The `CreatedAt` Search Criterion searches for payments based on the date when they were initiated.
```
Comment (dabrt) [inline docs/search/criteria_reference/payment_createdat_criterion.md:8]:
```suggestion
The `CreatedAt` Search Criterion searches for payments based on the date when they were created.
```

File: docs/search/criteria_reference/payment_search_criteria.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+Payment Search Criteria are part of [Search Criteria](search_criteria_reference.md) and they are supported only by Payment Search.
+
+With these Criteria you can filter payments by their payment identifier, payment creation date, payment status, payment method, order, and so on.
+
+## Payment Search Criteria
+
+|Search Criterion|Search based on|
+|-----|-----|
+|[CreatedAt](payment_createdat_criterion.md)|Date and time when payment was initiated|
```
Comment (dabrt) [inline docs/search/criteria_reference/payment_search_criteria.md:16]:
```suggestion
|[CreatedAt](payment_createdat_criterion.md)|Date and time when payment was created|
```

File: docs/search/sort_clause_reference/payment_createdat_sort_clause.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
+---
+description: Payment CreatedAt Sort Clause
+edition: commerce
+---
+
+# Payment CreatedAt Sort Clause
+
+The `CreatedAt` Sort Clause sorts search results by the date and time when the payment was initiated.
```
Comment (dabrt) [inline docs/search/sort_clause_reference/payment_createdat_sort_clause.md:8]:
```suggestion
The `CreatedAt` Sort Clause sorts search results by the date and time when the payment was created.
```

File: docs/search/sort_clause_reference/payment_sort_clauses.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+Payment Sort Clauses are part of [Sort Clauses](sort_clause_reference.md) and they are supported only by Payment Search.
+
+By using Sort Clauses you can filter payments by specific attributes, for example: creation date, status, and so on.
+
+| Sort Clause | Sorting based on |
+|-----|-----|
+|[Id](payment_id_sort_clause.md)|Payment ID|
+|[Identifier](payment_identifier_sort_clause.md)|Payment identifier|
+|[CreatedAt](payment_createdat_sort_clause.md)|Date and time when payment was initiated|
```
Comment (dabrt) [inline docs/search/sort_clause_reference/payment_sort_clauses.md:17]:
```suggestion
|[CreatedAt](payment_createdat_sort_clause.md)|Date and time when payment was created|
```

**Pull request merged**